File: 260. Single Number III.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        s = set()
        for i in nums:
            if i in s:
                s.remove(i)
            else:
                s.add(i)
        return list(s)

File: 260. Single Number III.cpp
#include <set>
class Solution {
public:
    vector<int> singleNumber(vector<int>& nums) {
        set <int> s;
        for (int i = 0; i < nums.size(); i++)
        {
            if (s.count(nums[i]) > 0)
                s.erase(nums[i]);
            else
                s.insert(nums[i]);
        }
        vector <int> res;
        copy(s.begin(), s.end(), back_inserter(res));
        return res;
    }
};
